tests/compatibility/WriteTillOverrideTestCase.test.nut

```diff
@@ -0,0 +1,67 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+@include __PATH__+"/../Core.nut"
+
+// WriteTillOverrideTestCase
+// Tests for SPIFlashLogger.write()
+class WriteTillOverrideTestCase extends Core {
+
+    function testBasic() {
+        return Promise(function(resolve, reject) {
+            try {
+                if (!isAvailable()) {
+                    resolve();
+                    return;
+                }
+                local start = 0;
+                local end = (start + 1) * SPIFLASHLOGGER_SECTOR_SIZE;
+                local logger = SPIFlashLogger(start, end, false, Serializer);
+                logger.erase();
+                local message = 1;
+                local messageDifferent = 2;
+                local messagesPerSector = SPIFLASHLOGGER_SECTOR_SIZE / Serializer.sizeof(message, SPIFLASHLOGGER_OBJECT_MARKER);
+                for (local i = 0; i < messagesPerSector + 1; i++) {
+                    if (i < messagesPerSector) {
+                        logger.write(message);
+                    } else {
+                        logger.write(messageDifferent);
```

Comment:
why isn't this throwing an error, while similar code in `WriteTooBigDataTestCase` does?

Reply:
?

Reply:
it is single huge object:
 WriteTooBigData check that object doesn't fit to the allocated sectors and throw an exception. I mean that object is bigger than the number of sectors that we have for logger. 

The number of small objects-
The current test case it checking data over-write when we have a lot of small objects